Fixes for animate docs

FRSgit opened the pull request:
<!-- General PR submission guidelines https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#submit-pr -->
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Updated animaton-aware directive tables both on ngAnimate's module and on animate guide pages.


**What is the current behavior? (You can also link to an open issue here)**
That's an effect of discussion which took place under #16561 


**What is the new behavior (if this is a feature change)?**
No, just docs.


**Does this PR introduce a breaking change?**
No.


**Please check if the PR fulfills these requirements**
- [x] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [x] Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated
- [x] Fix/Feature: Tests have been added; existing tests pass

**Other information**:

FRSgit committed:
docs(ngAnimate): refactor of ngAnimate docs

`Which directives support animations?` section:
Added info about ngAnimateSwap directive.
Sorted table alphabetically, now it's the same order as appear in documentation's navigation.

References #16561

FRSgit committed:
docs(ngAnimate): refactor of animation aware directives table

Synced this table with one which appears in animations guide (docs/content/guide/animations.ngdoc).

References #16561

FRSgit commented:
I signed it!

gkalpak reviewed (changes requested):
LGTM except for the (re-)ordering. My guess is that you tried to switch to some sort of alphabetic ordering, but it doesn't seem to have worked out very well 😛

FRSgit commented:
Yup, I've reordered it, so they appear in the same order as in navigation of the docs, what's bad about it?

FRSgit commented:
> Previously, they were grouped by animation type (e.g. structural vs non-structural), but not ordered other than that (afaict). Also, similar directives where close to each other (e.g. ngHide/ngShow).
Ah, I get it now. Well, then I'll put them alphabetically since it's the only obvious (for reader) way of handling it I guess (the *structural* naming is not so openly described in docs).
Still, the main focus there was to sync table from naAnimate page & animation guide.

FRSgit committed:
docs(ngAnimate): reorder animated directives tables

Reorder directives tables due to pull request's review (#16581)
Unify under-the-table informations.

Refers #16561
Refers #16581

FRSgit commented:
Tables reordered & unified.

gkalpak commented on src/ngAnimate/module.js:
```diff
@@ -17,20 +17,28 @@
+ * (More information can be found by visiting the documentation associated with each directive.)
+ *
+ * For a full breakdown of the steps involved during each animation event, refer to the
+ * {@link ng.$animate API docs}.
```
The text of the link will `API docs` (which is kind of vague). Maybe change it to `` `$animate` API docs``?

FRSgit replied:
Sure, just copied & pasted it from animations guide, wait for the commit ⌛️

gkalpak reviewed (approved):
LGTM (with one minor suggestion) 👍

FRSgit commented:
Done!
